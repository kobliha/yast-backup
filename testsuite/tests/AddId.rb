# encoding: utf-8

# Backup module - testsuite
#
# AddId function tests
#
# testedfiles: backup/functions.ycp
#
# $Id$
#
module Yast
  class AddIdClient < Client
    def main
      Yast.import "UI"

      Yast.import "Testsuite"

      @readmap = {
        "target" => { "string" => "", "size" => -1, "tmpdir" => "/tmp" },
        "proc"   => { "cpuinfo" => { "value" => { "0" => { "flags" => "" } } } },
        "probe"  => {
          "has_smp" => false,
          "is_uml"  => false,
          "cdrom"   => { "manual" => [] }
        }
      }
      @execmap = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stderr" => "",
            "stdout" => "charmap=\"ISO-8859-1\"\n"
          }
        }
      }

      Testsuite.Init([@readmap, {}, @execmap], nil)

      Yast.include self, "backup/functions.rb"

      Testsuite.Dump(AddId(nil))
      Testsuite.Dump(AddId([]))
      Testsuite.Dump(AddId(["/dev"]))
      Testsuite.Dump(AddId(["/dev", "/proc"]))

      nil
    end
  end
end

Yast::AddIdClient.new.main
