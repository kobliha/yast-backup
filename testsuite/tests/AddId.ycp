// Backup module - testsuite
//
// AddId function tests
//
// testedfiles: backup/functions.ycp
//
// $Id$
//

{

import "Testsuite";

map readmap = $[ "target" : $["string" : "", "size" : -1, "tmpdir" : "/tmp" ],
    "proc" : $["cpuinfo" : $[ "value" : $["0" : $[ "flags" : ""]]]],
    "probe" : $["has_smp" : false, "is_uml" : false, "cdrom" : $["manual" : []]],
    ];
map execmap = $[ "target" : $["bash_output" : $["exit":0, "stderr":"", "stdout":"charmap=\"ISO-8859-1\"\n"]]];

Testsuite::Init([ readmap, $[], execmap ], nil);

include "backup/functions.ycp";

Testsuite::Dump( AddId(nil) );
Testsuite::Dump( AddId([]) );
Testsuite::Dump( AddId(["/dev"]) );
Testsuite::Dump( AddId(["/dev", "/proc"]) );

}

