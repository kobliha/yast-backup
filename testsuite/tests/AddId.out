Dump	[]
Dump	[]
Dump	[`item (`id ("/dev"), "/dev")]
Dump	[`item (`id ("/dev"), "/dev"), `item (`id ("/proc"), "/proc")]
